### PR TITLE
LogBuddy integration

### PR DESCRIPTION
**Pull request by RigglePrime**
List of changes:

- LogBuddy integration
- LogBuddy has been improved
- Added RoundListLogDownloader
- Added get_round_json
- Slur detector no longer raises on load, that was annoying sometimes
- TG auth automatically copies the URL and pastes tokens
- Miscellaneous code changes

### CODE REVIEW

File: ss13_tools/log_buddy/log.py [hunk 1]
```diff
@@ -0,0 +1,558 @@
...
+        # NOTE: There is no better way of doing this. Why? Because the ckey isn't a ckey,
+        # it's a key WHICH COULD CONTAIN SPACES AND IT'S IMPOSSIBLE TO TELL WHAT IS PART
+        # OF THE KEY AND WHAT ISN'T. I love SS13 logs.
+        parse_key = False
+        other_temp = None
+        # One word
+        if other.startswith("injected"):
+            other_temp = other.split(" ", 1)[1]
+            parse_key = True
+        # NOTE: Performance? Not sure if it helps go check yourself, I am too lazy
```
Comment (dragomagol) [inline ss13_tools/log_buddy/log.py:724]: So scuffed....

File: ss13_tools/log_buddy/log.py [hunk 2]
```diff
@@ -0,0 +1,558 @@
...
+        else:
+            self.silicon_log_type = SiliconLogType.MISC
+        agent, other = log.split(") ", 1)
+        self.agent = Player.parse_player(agent)
+
+        if self.silicon_log_type == SiliconLogType.LAW and other.startswith("used "):
+            agent = other.split(" on ", 1)[1].split(") ", 1)[0]
+            self.agent = Player.parse_player(agent)
+        self.text = other
+        # NOTE: someone PLEASE fix logging this is getting ridiculous
```
Comment (dragomagol) [inline ss13_tools/log_buddy/log.py:840]: Make an issue report, goob

Comment (RigglePrime) [reply ss13_tools/log_buddy/log.py:840]: Issue report: logging is bad

Comment (dragomagol) [reply ss13_tools/log_buddy/log.py:840]: wtf

File: ss13_tools/menu.py [hunk 3]
```diff
@@ -55,7 +56,7 @@ def run(self):
 
 
 class RoundSingleItem(MenuItem):
-    weight = 3
+    weight = 4
     name = "round log downloader"
     description = "Download logs from a range of rouds"
```
Comment (dragomagol) [inline ss13_tools/menu.py]:
```suggestion
    description = "Download logs from a range of rounds"
```

**Pull request merged**